src/lexer/common.ts

```diff
@@ -28,26 +28,42 @@ export const enum NumberKind {
  */
 export function advanceChar(parser: ParserState): number {
   parser.column++;
-  return advanceAndLawrenceDolTheCRLF(parser);
-}
-
-export function advanceAndLawrenceDolTheCRLF(parser: ParserState) {
   parser.currentChar = parser.source.charCodeAt(++parser.index);
-  if (parser.index < parser.end) {
-    const cur = parser.currentChar;
-    const nxt = parser.source.charCodeAt(parser.index + 1);
-    if (
-      (cur == Chars.CarriageReturn && nxt == Chars.LineFeed) ||
-      (cur == Chars.LineFeed && nxt == Chars.CarriageReturn)
-    ) {
-      parser.currentChar = Chars.LineFeed;
-      parser.index++;
-    } else if (cur === Chars.CarriageReturn) {
-      parser.currentChar = Chars.LineFeed;
-    }
+
+  if(parser.index < parser.end) {
+      const cur = parser.currentChar;
+      const nxt = parser.source.charCodeAt(parser.index + 1);
+      if((cur == Chars.CarriageReturn && nxt == Chars.LineFeed) || (cur == Chars.LineFeed && nxt == Chars.CarriageReturn)) {
```

Comment:
Can't you use a second func arg that validates to true if CR. That way you could simplify and optimize this line?

Reply:
And you should always use `===` and not `==`

Reply:
This is testing for a `CRLF` or `LFCR` sequence, not just CR. Starting from the beginning any LF, CR, CRLF, or LFCR must be treated as a line-break. This logic works for a file which is entirely one of these, or a mix. 

Now, in practice LFCR was never used by any OS though it's technically allowed instead of CRLF, lone CR was used only by Mac OS9 and earlier, and today only LF, and CRLF really exist in practice. Depending on git settings and editors it's quite possible for an editor to emit a mixed LF & CRLF file. This logic copes with all combinations, and is simple enough.

To cope with only LF and CRLF (or a mix of them), the above check for `(cur === LF && nxt === CR)` could be dropped.

I don't see a subfunction being sufficiently clearer to warrant the one-liner, but if you want I will:

    function isLineBreakPair(cur,nxt) {
        return (cur == Chars.CarriageReturn && nxt == Chars.LineFeed) || (cur == Chars.LineFeed && nxt == Chars.CarriageReturn);
    }

Reply:
OK, I did that.